src/setup-ghidra-source.cmake

```diff
@@ -0,0 +1,62 @@
+# ---- Setup Ghidra Source code ----
+
+# Set up Ghidra repo human-readable version settings
+set(SLEIGH_GHIDRA_RELEASE_TYPE "stable" CACHE
+  STRING "Ghidra release type to use. Make sure SLEIGH_GHIDRA_COMMIT is the correct corresponding commit. 'HEAD' is used for active development purposes."
+)
+# This is just helper for CMake UIs. CMake does not enforce that the value matches one of those listed.
+set_property(CACHE SLEIGH_GHIDRA_RELEASE_TYPE PROPERTY STRINGS "stable" "HEAD")
+
+# **** Setup pinned git info ****
+
+# Ghidra pinned stable version commit
+set(ghidra_version "10.0.4")
+set(ghidra_git_tag "5b07797cb859d8801ca3a1e08bda1321ca3ce002")
+# pinned stable patches list
+set(ghidra_patches "")
+
+# Ghidra pinned commits used for pinning last known working HEAD commit
```

Comment:
Do we have any other builds that work like this? I'm just wondering how we usually manage `HEAD` in a setup like this. Maybe I should periodically bump it to the tip of `master` and make sure it builds ok.

Reply:
Honestly, I'm not aware of any other projects doing something like this.

I think some sort of bot or GitHub Action would be the best way to keep track of this. Sort of like Dependabot but we would have to script it ourselves. I see the workflow like:

1. Create a weekly GitHub Action cron job
2. Grab the latest commit from default development Ghidra branch (`master`)
3. Check that it isn't the same commit as what we have currently
4. Write new commit SHA to the `./src/setup-ghidra-source.cmake` HEAD commit variable
5. Commit and push as new PR
6. Let CI run
7. Wait for manual approval

It would also be nice to automatically update the listing of `slaspec` files...

Reply:
Thanks. That sounds like a good idea.